fix docs for `format_source`

bernt-matthias opened the pull request:
Document that reference to parameters in `format_source` in sections and conditionals needs to be qualified with `|`. Collection elements can be referred to using element access syntax.

To verify this I added a log statement here: lib/galaxy/tools/actions/__init__.py printing `format_source`, `input_datasets` and `input_dataset_collections`. Here we look at a test for a tool with a pair `paired_input` in a conditional `single_paired` (IUC's fastp tool). Access would be via `'single_paired|paired_input`: 

```
galaxy.tools.actions ERROR 2024-12-16 16:14:29,475 [pN:main.1,p:2622874,tN:WSGI_0] format_source=None input_datasets={'single_paired|paired_input1': <galaxy.model.HistoryDatasetAssociation(65) at 0x7f7c610de860>, 'single_paired|paired_input2': <galaxy.model.HistoryDatasetAssociation(66) at 0x7f7c610dd000>} input_dataset_collections={'single_paired|paired_input': [(<galaxy.model.HistoryDatasetCollectionAssociation(16) at 0x7f7c60e45fc0>, False)]}
```

For the single ended input this looks analogous. Bottom line the dictionaries do not contain the unqualified inputs. In the  fastp tool additional output actions were implemented https://github.com/galaxyproject/tools-iuc/blob/a80e3e4aa3a40970af507bf9119cf7f1c2ffb336/tools/fastp/macros.xml#L73 .. likely because the unqualified [`format_source`](https://github.com/galaxyproject/tools-iuc/blob/a80e3e4aa3a40970af507bf9119cf7f1c2ffb336/tools/fastp/fastp.xml#L357) did not work.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

bernt-matthias committed:
fix docs for format_source

Reference to parameters in sections and conditionals needs to be qualified
with `|`. Collection elements can be referred to using element access
syntax.

mvdbeek commented:
Seems kind of weird to have documented wrong examples ? Are you sure those specific examples are broken and it's not a general issue with tool profile versions or actual bugs ?

bernt-matthias commented:
> Seems kind of weird to have documented wrong examples ? Are you sure those specific examples are broken and it's not a general issue with tool profile versions or actual bugs ?

I'm not sure. I could add a test checking if it works currently.

bernt-matthias committed:
fix test and improve for legacy mapping

the test for the nesting case accidentally works since the
[legacy_mapping](https://github.com/galaxyproject/galaxy/blob/16ec912385b59429b1e87cebb817ec012b02ec4e/lib/galaxy/tools/parameters/wrapped.py#L43) contains a wrong mapping.
with the output2 we can now test this.

with output3 we now also test that the legacy behavior works.

bernt-matthias committed:
temporary debug code

bernt-matthias committed:
more temporary debug code

bernt-matthias committed:
fix debug code

bernt-matthias committed:
fix test

mvdbeek commented:
Can you resolve the conflicts and remove the comments of your log statements if you want this in for 25.0 ?

bernt-matthias committed:
Merge branch 'dev' into topic/fix_format_source_docs
